framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -707,10 +707,13 @@ public AnnotationMirror getAnnotationFromJavaExpressionString(
 
         AnnotationMirror annotationMirror = null;
         if (CFAbstractStore.canInsertReceiver(expressionObj)) {
-            Value value = getStoreBefore(tree).getValue(expressionObj);
-            if (value != null) {
-                annotationMirror =
-                        AnnotationUtils.getAnnotationByClass(value.getAnnotations(), clazz);
+            Store store = getStoreBefore(tree);
```

Comment:
This should be in a separate PR with a test case.  It's a general bug fix to a framework class.